agent/exploits/cve_2024_43044.py

```diff
@@ -0,0 +1,81 @@
+"""Agent Asteroid implementation for CVE-2024-43044"""
+
+import re
+from packaging import version
+from requests import exceptions as requests_exceptions
+import requests
+from urllib import parse as urlparse
+
+from agent.exploits import webexploit
+from agent import exploits_registry
+from agent import definitions
+
+
+VULNERABILITY_TITLE = "JENKINS ARBITRARY FILE READ TO RCE"
+VULNERABILITY_REFERENCE = "CVE-2024-43044"
+VULNERABILITY_DESCRIPTION = """Arbitrary file read vulnerability through agent connections can lead to RCE 
+"""
+RISK_RATING = "CRITICAL"
+# todo update the version to test also for LTS
+MAX_VULNERABLE_LTS_VERSION = version.parse("2.452.3")
+MAX_VULNERABLE_VERSION = version.parse("2.470")
+
+MAX_REDIRECTS = 2
+DEFAULT_TIMEOUT = 90
+
+
+@exploits_registry.register
+class CVE202443044Exploit(webexploit.WebExploit):
+    accept_request = definitions.Request(method="GET", path="/")
+    check_request = definitions.Request(method="GET", path="/")
+    accept_pattern = [re.compile("Sign in \[Jenkins\]")]
+    metadata = definitions.VulnerabilityMetadata(
+        title=VULNERABILITY_TITLE,
+        description=VULNERABILITY_DESCRIPTION,
+        reference=VULNERABILITY_REFERENCE,
+        risk_rating=RISK_RATING,
+    )
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        """Rule to detect specific vulnerability on a specific target.
+
+        Args:
+            target: Target to scan
+
+        Returns:
+            List of identified vulnerabilities.
+        """
+        session = requests.Session()
+        session.max_redirects = MAX_REDIRECTS
+        session.verify = False
+
+        vulnerabilities: list[definitions.Vulnerability] = []
+
+        target_endpoint = urlparse.urljoin(target.origin, self.check_request.path)
+
+        try:
+            req = requests.Request(
+                method=self.check_request.method,
+                url=target_endpoint,
+                data=self.check_request.data,
+            ).prepare()
+            if self.check_request.headers is not None:
+                self.check_request.headers.update(req.headers)
+                req.headers = self.check_request.headers  # type: ignore
+            resp = session.send(req, timeout=DEFAULT_TIMEOUT, verify=False)
```

Comment:
do u need `verify=False` if u already set it on line 50?

Reply:
no need

Reply:
this one is from the base class webexploit i just override the version check